api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,486 @@
+        '503':
+          description: Not available
+          schema:
+            $ref: '#/definitions/Problem'
+  '/topics/{topic}/events':
+    get:
+      tags:
+        - low-level-api
+      summary: |
+        Starts a stream delivery for the specified partitions of the given topic.
+        The tracking of the current position in the partitions and of which
+        partitions is being read is in the responsibility of the client.
+        No commits are needed.
+      parameters:
+        - name: topic
+          in: path
+          description: Topic where to get events from
+          type: string
+          required: true
+        - name: x-nakadi-cursors
+          in: header
+          description: Cursors pointing to partitions/offsets to start read from
+          required: false
+          type: array
+          items:
+            type: string
+            format: '#/definitions/Cursor'
+        - name: batch_limit
+          in: query
+          description: Maximum number of `SimpleStreamElement`s in each chunk of the stream. If unspecified assumes default value 1 (i.e. each event is individually submitted).
+          type: integer
+          format: int32
+          required: false
+          default: 1
+        - name: stream_limit
+          in: query
+          description: |
+            Maximum number of `Event`s in this stream. If 0 or undefined, will
+            stream indefinately.
+
+            Stream initialization will fail if `stream_limit` is lower than
+            `batch_limit`.
+          type: integer
+          format: int32
+          required: false
+        - name: batch_flush_timeout
+          in: query
+          description: Maximum time in seconds to wait for the flushing of each chunk; if the `batch_limit` is reached before this time is reached the messages are immediately flushed to the client.
+          type: integer
+          format: int32
+          required: false
+        - name: stream_timeout
+          in: query
+          description: |
+            Maximum time in seconds a stream will live before being interrupted.
+            If unspecified will stream indefinately.
+
+            If this timeout is reached any pending messages (in the sense of
+            `stream_limit`) will be flushed to the client.
+
+            Stream initialization will fail if `stream_timeout` is lower than
+            `batch_flush_timeout`.
+          type: integer
+          format: int32
+          required: false
+        - name: batch_keep_alive_limit
+          in: query
+          description: Maximum number of keep-alive messages to get in a row before closing the connection. Unlimited by default.
+          type: integer
+          format: int32
+          required: false
+      responses:
+        '200':
+          description: |
+            Starts streaming to the client.
+            Stream format is a continuous series of
+            `SimpleStreamEvent`s separated by `\n`
```

Comment:
Line delimited json (https://en.wikipedia.org/wiki/JSON_Streaming#Line_delimited_JSON) might be easier to deal with in a streaming fashion compared to arrays, at least for some parser implementation. The problem I see is that there is no standard content type for it, but I saw several mentions of using `application/x-json-stream`.